src/behaviors/StopBehavior.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\queue\behaviors;
+
+use yii\base\Behavior;
+use yii\caching\Cache;
+use yii\di\Instance;
+use yii\queue\ExecEvent;
+use yii\queue\Queue;
+
+/**
+ * Stoppable behavior allow stopping scheduled jobs in a queue.
+ *
+ * This behavior provides a [[stop()]] method, which allows to mark scheduled jobs as "stopped", which
+ * will prevent their execution.
+ *
+ * This behavior should be attached to the [[Queue]] component.
+ *
+ * @author Roman Zhuravlev <zhuravljov@gmail.com>
+ * @since 2.0.1
+ */
+class StopBehavior extends Behavior
+{
+    /**
+     * @var Cache|array|string the cache instance used to store stopped status.
+     */
+    public $cache = 'cache';
+    /**
+     * @var bool
+     */
+    public $checkWaiting = true;
+    /**
+     * @var Queue
+     * @inheritdoc
+     */
+    public $owner;
+
+    /**
+     * @inheritdoc
+     */
+    public function init()
+    {
+        parent::init();
+        $this->cache = Instance::ensure($this->cache, Cache::class);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function events()
+    {
+        return [
+            Queue::EVENT_BEFORE_EXEC => function (ExecEvent $event) {
```

Comment:
making this a `beforeExec()` method instead of a closure could make it easier to extend this class.